packages/typespec-azure-resource-manager/lib/common-types/managed-identity.tsp

```diff
@@ -33,10 +33,24 @@ model ManagedServiceIdentity {
   type: ManagedServiceIdentityType;
 
   @doc("The identities assigned to this resource by the user.")
-  userAssignedIdentities?: UserAssignedIdentities;
+  userAssignedIdentities?: Record<UserAssignedIdentity>;
 }
 
 /** The set of user assigned identities associated with the resource. The userAssignedIdentities dictionary keys will be ARM resource ids in the form: '/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}. The dictionary values can be empty objects ({}) in requests.", */
+#deprecated "Please use `Record<UserAssignedIdentity>` directly for better SDK API."
```

Comment:
Added deprecation notice.

Reply:
```suggestion
#deprecated "DO NOT USE, use `Record<UserAssignedIdentity>` directly, using this model will result in a different client SDK when generated from TypeSpec vs the swagger output."
```
can we make the message clearer that this should not be used at all? It nost just that it will give better SDK, using it will result in a breaking change when generation goes from swwgger to typespec